### ISSUE TITLE
With too many apb's bootstrap causes the route to timeout.

### ISSUE THREAD

**Issue by jmontleon** [openshift/ansible-service-broker#876]
## Bug:
Pretty low priority at this point.

With too many apb's bootstrap causes the route to timeout.

**What happened**:
```
apb bootstrap
Contacting the ansible-service-broker at: https://asb-1338-ansible-service-broker.172.17.0.1.nip.io/ansible-service-broker/v2/bootstrap
Error: Attempt to bootstrap Broker returned status: 504
Unable to bootstrap Ansible Service Broker.
```

**What you expected to happen**:
Boostrap completes.

**How to reproduce it**:
Populate a whole lot of APB's. In my cases I created 150 service templates in CFME while running a simple test of the adapter. Those were in addition to the ones available at docker.io/ansibleplaybookbundle that were also being loaded.

Boostrap actually completes but the route times out. I'd guess the defaults is ~10-30 seconds. My boostrap is taking probably closer to a minute

`oc annotate route -n ansible-service-broker asb-1338 --overwrite haproxy.router.openshift.io/timeout=300s` stops it from happening.

We could probably add the annotation to the route we create with a reasonable setting higher than the default, otherwise as we see users with more than a handful of APB's we may start seeing folks reporting this.

**Comment by SaravanaStorageNetwork** [comment]
@rthallisey  Thanks!  It helped - able to run bootstrap only after extending the timeout as mentioned.

**Comment by jwmatthews** [comment]
Assuming correct fix here is to make this an async call, thoughts?
If we wanted to make this async assume we align this to 3.11 to limit churn in 3.10.

**Comment by jmrodri** [comment]
@jwmatthews yes if the bootstrap is taking too long, we should spawn it off as we start.

**Comment by jmontleon** [comment]
@jwmatthews @jmrodri when running 'apb boostrap' it's waiting for the bootstrap to complete to also do a relist so the catalog sees the updates as well.

If it was changed to async would that behavior have to be changed in the client tool?

**Comment by jmrodri** [comment]
@jmontleon I misread the original complaint about `apb bootstrap`. I thought it was during broker startup. hrm that makes it a bit more complicated. Making it async would mean the bootstrap call returns immediately but we'd need a way to get back the same information.

**Comment by jmontleon** [comment]
We could probably annotate the route we create with a 1 or 2 minute timeout for now. I think it would cover most cases for the time being. If we ever see cases of someone with 100's of APB's we might have to readdress it, but even close to 200 was returning for me in under a minute.

**Comment by jwmatthews** [comment]
Example @jmontleon shared:
```
oc annotate route -n ansible-service-broker asb-1338 --overwrite haproxy.router.openshift.io/timeout=300s
```

**Comment by jmrodri** [comment]
Fixed by PR #1008

**Comment by wenchma** [comment]
@jwmatthews  it's unlucky,  the 504 error not gone after increasing the timeout as your mentioning .
😭

**Comment by jmontleon** [comment]
There are a couple PR's hanging out to fix this. Trying to get them merged:
https://github.com/automationbroker/automation-broker-apb/pull/27
https://github.com/openshift/ansible-service-broker/pull/1033

@wenchma can you describe the asb route and verify the annotation is there? Is it actually taking 5 minutes (300s) to return now if you set that value? If that doesn't seem right maybe we're hitting something else.

**Comment by wenchma** [comment]
@jmontleon  thanks for your reply, I guess I need a longer timeout.

**Comment by jmontleon** [comment]
As mentioned it there might be something else happening that we haven't run into before as well. 5 minutes seems like an awfully long time to bootstrap unless you are loading a huge number of APB's (I managed to get to around 60s by creating a couple hundred APB's)

**Comment by wenchma** [comment]
@jmontleon  it looks like that the 5 minutes timeout did not take effect.

**Comment by jmrodri** [comment]
Closing as this has been fixed.
